test/client_test.dart

```diff
@@ -116,22 +124,27 @@ void main() {
       expect(data?.user.userMetadata, {'Hello': 'World'});
     });
 
-    test('signUp() with autoConfirm off with email', () async {
-      final response = await clientWithAuthConfirmOff.signUp(
-        email: email,
-        password: password,
-        emailRedirectTo: 'https://localhost:9999/welcome',
-      );
-      expect(response.user, isA<User>());
-      expect(response.session, isNull);
+    test(
```

Comment:
This test was VERY strange for me. When actually trying to send the confirmation email I get a 500 server error, but when the account already existed with phone (as it was before) I don't get any error. I changed it now to expect a fail like the phone confirmation. Don't really know what's going on here.

Reply:
Yeah, this behavior does look very strange. It would still be nice to be able to confirm that calling `signUp()` with autoConfirm off does return a user and no session though. Let me ask the auth team if there is a way to test it out without gotrue throwing an error.

Reply:
Or instead we mock the response with a custom http client?

Reply:
@Vinzent03 That would work too! Let's do that!

Reply:
@dshukertjr Should be fixed now.